common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -60,18 +63,26 @@ public void recycle(Object object) {
         if (maxCapacity < 0) {
             maxCapacity = DEFAULT_INITIAL_MAX_CAPACITY;
         }
-
         DEFAULT_MAX_CAPACITY = maxCapacity;
 
+        int maxSharedCapacityFactor = SystemPropertyUtil.getInt("io.netty.recycler.maxSharedCapacityFactor",
+                DEFAULT_MAX_SHARED_CAPACITY_FACTOR);
+        if (maxSharedCapacityFactor < 0) {
+            maxSharedCapacityFactor = DEFAULT_MAX_SHARED_CAPACITY_FACTOR;
+        }
+        MAX_SHARED_CAPACITY_FACTOR = maxSharedCapacityFactor;
```

Comment:
unless its intentional that `[0, DEFAULT_MAX_SHARED_CAPACITY_FACTOR)` should stay as is while negative values should be set to `DEFAULT_MAX_SHARED_CAPACITY_FACTOR` ... if so can you add a comment?